grouper/ctl/util.py

```diff
@@ -35,42 +36,42 @@ def wrapper(args):
         if not valid:
             return
 
-        return f(args)
+        return f(args, settings)
 
     return wrapper
 
 
 def ensure_valid_groupname(f):
-    # type: (Callable[[Namespace], None]) -> Callable[[Namespace], None]
+    # type: (Callable[[Namespace, CtlSettings], None]) -> Callable[[Namespace, CtlSettings], None]
     @wraps(f)
-    def wrapper(args):
-        # type: (Namespace) -> None
+    def wrapper(args, settings):
+        # type: (Namespace, CtlSettings) -> None
         if not re.match("^{}$".format(NAME_VALIDATION), args.groupname):
             logging.error("Invalid group name {}".format(args.groupname))
             return
 
-        return f(args)
+        return f(args, settings)
 
     return wrapper
 
 
 def ensure_valid_service_account_name(f):
-    # type: (Callable[[Namespace], None]) -> Callable[[Namespace], None]
+    # type: (Callable[[Namespace, CtlSettings], None]) -> Callable[[Namespace, CtlSettings], None]
     @wraps(f)
-    def wrapper(args):
-        # type: (Namespace) -> None
+    def wrapper(args, settings):
+        # type: (Namespace, CtlSettings) -> None
         if not re.match("^{}$".format(SERVICE_ACCOUNT_VALIDATION), args.name):
             logging.error('Invalid service account name "{}"'.format(args.name))
             return
 
-        return f(args)
+        return f(args, settings)
 
     return wrapper
 
 
-def make_session():
-    # type: () -> Session
-    db_engine = get_db_engine(get_database_url(settings()))
+def make_session(settings):
+    # type: (Settings) -> Session
```

Comment:
What's the decision making process for using a generic `Settings` versus a more specific subclass (`CtlSettings` here)? To me it feels like since this is only called within grouper/ctl, this should be `CtlSettings`, but can also see an argument for making it more generic since no ctl-specific settings are used.

Reply:
Yeah, good call, this should be `CtlSettings` as well. Will fix that and the other case.